File: old/arrHM/001-twoNumberSum.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

vector<int> twoNumberSum(vector<int>& arr, int target) {
    vector<int> ans;
    unordered_set<int> seen;
    for (int num: arr) {
        if (seen.count(target - num) != 0) {
            ans.push_back(min(num, target-num));
            ans.push_back(max(num, target-num));
        }
        seen.insert(num);
    }
    return ans;
}

int main() {
	//code
	int T;
	cin >> T;
	
	while (T--) {
	    
	    int n;
	    cin >> n;
	    
	    int target;
	    cin >> target;

	    
	    vector<int> v(n);
	    for (int i = 0; i < n; i++) {
	        cin >> v[i];
	    }
	    
	    
	    vector<int> ans = twoNumberSum(v, target);
	    
	    if (ans.size() != 0) {
	        cout << "Yes" << endl;
	    } else {
	        cout << "No" << endl;
	    }
	}
	return 0;
}

File: old/arrHM/001-twoNumberSum.java
/*package whatever //do not write package name here */

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
    
    public static List<Integer> twoNumberSum(List<Integer> arr, int target) {
        Set<Integer> seen = new HashSet<>();
        List<Integer> ans = new ArrayList<>();
        for (int num: arr) {
            if (seen.contains(target - num)) {
                 ans.add(num); 
                 ans.add(target-num);
                 return ans;
            }
            seen.add(num);
        }
        return ans;
    }
    
	public static void main (String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		while ((T--) != 0) {
	    
    	    int n;
    	    n = sc.nextInt();
    	    
    	    int target;
    	    target = sc.nextInt();
    
    	    
    	    List<Integer> v = new ArrayList<>();
    	    for (int i = 0; i < n; i++) {
    	        v.add(sc.nextInt());
    	    }
    	    
    	    
    	    List<Integer> ans = twoNumberSum(v, target);
    	    
    	    if (ans.size() != 0) {
    	        System.out.println("Yes");
    	    } else {
    	        System.out.println("No");
    	    }
	    }

	}
}